steam/ext/dota2/client.py

```diff
@@ -0,0 +1,127 @@
+from ...app import DOTA2
+from ...ext import commands
+from ...utils import MISSING
+from .protobufs.dota_gcmessages_client_watch import (
+    CMsgClientToGCFindTopSourceTVGames,
+    CMsgGCToClientFindTopSourceTVGamesResponse,
+)
+from .state import GCState  # noqa: TCH001
+
+__all__ = (
+    "Client",
+    "Bot",
+)
+
+
+class Client(Client_):
+    """Represents a client connection that connects to Steam. This class is used to interact with the Steam API, CMs
+    and the Dota 2 Game Coordinator.
+
+    :class:`Client` is a subclass of :class:`steam.Client`, so whatever you can do with :class:`steam.Client` you can
+    do with :class:`Client`.
+    """
+
+    _APP: Final = DOTA2
+    _state: GCState  # type: ignore  # PEP 705
+
+    async def fetch_top_source_tv_games(
+        self,
+        *,
+        search_key: str = MISSING,
+        league_id: int = MISSING,
+        hero_id: int = MISSING,
+        start_game: Literal[0, 10, 20, 30, 40, 50, 60, 70, 80, 90] = 0,
+        game_list_index: int = MISSING,
+        lobby_ids: list[int] = MISSING,
+    ) -> list[CMsgGCToClientFindTopSourceTVGamesResponse]:
+        """Fetch Top Source TV Games.
+
+        This functionality is similar to game list in the Watch Tab of Dota 2 game application.
+        It fetches summary data for the currently on-going Dota 2 matches from the following categories:
+
+        * tournament games
+        * highest average mmr games
+        * specific lobbies, like friends' games by their lobby_ids
+
+        Note
+        -------
+        Note that the following documentation for keyword arguments to query against is rather observant
+        than from official. So, please, if you know more or description is incorrect, contact us.
+
+        Parameters
+        ----------
+        search_key : :class: `str`, optional
+            Unknown purpose.
+        league_id : :class: `int`, optional
+            `league_id` for the professional tournament.
+        hero_id : :class: `int`, optional
+            `hero_id` to filter results by, just like in-client Watch Tab feature.
+        start_game : :class: `Literal[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]`, optional, by default 0
+            This argument controls how many responses the game coordinator should return.
+            For example, `start_game=0` returns a list with 1 Response,
+            `start_game=90` returns a list of 10 Responses.
+            Game coordinator fills each response in consequent manner with games that satisfy keyword arguments
+            or with currently live highest average MMR games in case no other argument was given.
+        game_list_index : :class: `int`, optional
+            Only get responses matching `game_list_index`. Responses from Game Coordinator change
+            from time to time. The `game_list_index` indicates that those responses belong to the same chunk.
+        lobby_ids : :class: `list[int]`, optional
+            Lobby ids to query against.
```

Comment:
```suggestion
        league_id
            The league ID for the professional tournament.
        hero_id
            `hero_id` to filter results by which can be found in the client Watch Tab feature.
        start_game
            This argument controls how many responses the game coordinator should return.
            For example, `start_game=0` returns a list with 1 Response,
            `start_game=90` returns a list of 10 Responses.
            The GC fills each response in consequent manner with games that satisfy keyword arguments
            or with currently live highest average MMR games in case no other argument was given.
        game_list_index
            Only get responses matching `game_list_index`. Responses from GC change
            from time to time. The `game_list_index` indicates that those responses belong to the same chunk.
        lobby_ids
            The lobby IDs of the games to fetch.
```
The functionality for start_game doesn't really make much sense, would it be possible to have this just be a limit argument and it does whatever maths it needs to for you?

Reply:
actually, it kinda works like that, I guess i will rewrite this sentence, 

i'm not sure if I want to rename the param name - as I said, I kinda thought it would be nice to have it 1x1 matching valve docs. But yeah to something like `total_games` would make much more sense for what this thing is for.